qutebrowser/browser/hints.py

```diff
@@ -130,7 +144,38 @@ def _move_to_elem(self):
     def cleanup(self):
         """Clean up this element and hide it."""
         self.hide()
-        self.deleteLater()
+        self.setParent(None)
+        self._context.tab.contents_size_changed.disconnect(self._move_to_elem)
+        HintLabel._object_pool.append(self)
+
+        if len(HintLabel._object_pool)*2 > HintLabel._object_count:
```

Comment:
I'm not sure how I feel about this. I'm guessing that because of this, if I open a page with a lot of links, then browse around normally, and come back to hint, I'll experience more lag than before. On the other hand, it will save memory. I'm personally fine with loosing ~1MB to hints, as that won't scale by the number of tabs (at which point webengine will dominate memory usage).

I'm wondering what others will think about it.